client.go

```diff
@@ -154,23 +93,26 @@ func (c *Castle) sendRiskCall(ctx context.Context, e *castleAPIRequest) (Recomme
 	defer res.Body.Close() // nolint: gosec
 	if res.StatusCode != http.StatusCreated {
 		b, _ := io.ReadAll(res.Body) // nolint: errcheck
-		return RecommendedActionNone, errors.Errorf("expected 201 status but got %d: %s", res.StatusCode, string(b))
+		return RecommendedActionNone, fmt.Errorf("expected 201 status but got %d: %s", res.StatusCode, string(b))
 	}
 
 	resp := &castleAPIResponse{}
 	if err = json.NewDecoder(res.Body).Decode(resp); err != nil {
-		return RecommendedActionNone, errors.Errorf("unable to decode response body: %v", err)
+		return RecommendedActionNone, fmt.Errorf("unable to decode response body: %w", err)
 	}
 
-	if resp.Type != "" {
-		// we have an api error
-		return RecommendedActionNone, errors.New(resp.Type)
-	}
+	return recommendedActionFromString(resp.Policy.Action), nil
+}
 
-	if resp.Message != "" {
-		// we have an api error
-		return RecommendedActionNone, errors.Errorf("%s: %s", resp.Type, resp.Message)
+func recommendedActionFromString(action string) RecommendedAction {
+	switch action {
+	case "allow":
+		return RecommendedActionAllow
+	case "deny":
+		return RecommendedActionDeny
+	case "challenge":
+		return RecommendedActionChallenge
+	default:
+		return RecommendedActionNone
 	}
```

Comment:
Just double checking - do we no longer need/want to return errors in these cases, and `RecommendedActionNone` is sufficient?

Reply:
These two checks are redundant. We check for error above by looking at the http.StatusCode. They seem to me to be some left-overs.